runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -102,7 +102,8 @@ local function diagnostic_lsp_to_vim(diagnostics, bufnr, client_id)
       end_lnum = _end.line,
       end_col = line_byte_from_position(buf_lines, _end.line, _end.character, offset_encoding),
       severity = severity_lsp_to_vim(diagnostic.severity),
-      message = diagnostic.message
+      message = diagnostic.message,
+      code = diagnostic.code,
```

Comment:
Should we copy the other fields as well? e.g. tags, ...

See https://microsoft.github.io/language-server-protocol/specification.html#diagnostic for all possible official fields

Reply:
I think that should be fine.

@neovim/lsp any thoughts here?

Reply:
Should we maybe store all of the original raw lsp fields under an `lsp` key?

Reply:
I'd also tend to have some arbitary property bag and within that an `lsp` key, so that we can avoid extending the neovim specific diagnostic structure each time the LSP specification is extended. (Not that it happens often, but to keep it decoupled)